src/lib/route-builder.js

```diff
@@ -110,8 +110,8 @@ class RouteBuilder {
     const _nextHopBySourceAmount = this.routingTables.findBestHopForSourceAmount(
       sourceLedger, finalLedger, sourceTransfer.amount)
     if (!_nextHopBySourceAmount) throwAssetsNotTradedError()
-    // Round against ourselves since the quote rate was overestimated in our favor.
-    const nextHop = yield this._roundHop(_nextHopBySourceAmount, 'down', 'up')
+    // Round in favor of the connector.
+    const nextHop = yield this._roundHop(_nextHopBySourceAmount, 'up', 'down')
```

Comment:
I'll add more comments. The short answer as to why it changed: before it rounded in favor of the sender/receiver because the route shifting had already occurred, so despite the rounding the result was still overall in favor of the connector. Now, though, the local part of the route hasn't been shifted, so in order to avoid losing money the connector must round in its own favor.